### ISSUE TITLE
Nondeterministic specs

### ISSUE THREAD

**Issue by dariusf** [Coursemology/coursemology2#420]
Looks like the announcement specs are not completely deterministic yet.

First one happened twice today, second happened once.

```
Failures:

  1) Course: Announcements Pagination with tenant :instance after clicked second page lists each announcement
     Failure/Error: visit course_announcements_path(course)
     Bullet::Notification::UnoptimizedQueryError:
       user: darius
       Unused Eager Loading detected
         Course::Announcement => [:creator]
         Remove from your finder: :includes => [:creator]
     # ./spec/features/course/announcement_pagination_spec.rb:18:in `block (4 levels) in <top (required)>'
```

```
Failures:

   1) System: Administration: Announcements Pagination
      Failure/Error: System::Announcement.delete_all
      ActiveRecord::StatementInvalid:
        PG::TRDeadlockDetected: ERROR:  deadlock detected
        DETAIL:  Process 14068 waits for ShareLock on transaction 604847; blocked by process 14043.
        Process 14043 waits for ShareLock on transaction 604867; blocked by process 14068.
        HINT:  See server log for query details.
        CONTEXT:  while deleting tuple (5,36) in relation "generic_announcements"
        : DELETE FROM "generic_announcements" WHERE "generic_announcements"."type" IN ('System::Announcement')
      # ./spec/features/system/admin/announcement_pagination_spec.rb:10:in `block (3 levels) in <top (required)>'

   2) System: Administration: Announcements Pagination after clicked second page lists each announcement
      Failure/Error: System::Announcement.delete_all
      ActiveRecord::StatementInvalid:
        PG::TRDeadlockDetected: ERROR:  deadlock detected
        DETAIL:  Process 14068 waits for ShareLock on transaction 604847; blocked by process 14043.
        Process 14043 waits for ShareLock on transaction 604874; blocked by process 14068.
        HINT:  See server log for query details.
        CONTEXT:  while deleting tuple (3,46) in relation "generic_announcements"
        : DELETE FROM "generic_announcements" WHERE "generic_announcements"."type" IN ('System::Announcement')
      # ./spec/features/system/admin/announcement_pagination_spec.rb:10:in `block (3 levels) in <top (required)>'
```

**Comment by lowjoel** [comment]
I think we fixed some of these in #225.

**Comment by weiqingtoh** [comment]
Solved a long time back. Will close.
